### PR TITLE
Fix Safari inverting exported textures

### PR DESCRIPTION
**Pull request by Drigax**
Fix for issue where Safari canvas is drawn Y down, workaround was discovered where premultiplying alpha causes correct canvas orientation. Added utility method to detect Safari browser to use this alternate engine initializing path.

Addressing #7352

### CODE REVIEW

File: serializers/src/glTF/2.0/glTFExporter.ts [hunk 1]
```diff
@@ -335,7 +335,11 @@ export class _Exporter {
             localCanvas.id = "WriteCanvas";
             localCanvas.width = 2048;
             localCanvas.height = 2048;
-            this._localEngine = new Engine(localCanvas, true, { premultipliedAlpha: false, preserveDrawingBuffer: true });
+            if (Tools.IsSafari()) {
```
Comment (Drigax) [inline serializers/src/glTF/2.0/glTFExporter.ts]: @deltakosh, I noticed in my local build that I'm not able to "see" the new Tools utility method I added, causing the build to fail here. Is there anything I need to do to make sure that this method is exported properly?

File: src/Misc/tools.ts [hunk 2]
```diff
@@ -1215,6 +1215,14 @@ export class AsyncLoop {
             }
         }, callback);
     }
+
+    /**
+     * Utility function to detect if the current user agent is Safari
+     * @returns whether or not the current user agent is safari
+     */
+    public static IsSafari() : boolean {
+        return /^((?!chrome|android).)*safari/i.test(navigator.userAgent);
```
Comment (Drigax) [inline src/Misc/tools.ts]: kudos of some logic in the thingine that is used to detect certain bad desktop OSes. 😉

I opted to add this logic to Tools as well, but perhaps I should modify thinEngine to reuse the logic there instead?

File: serializers/src/glTF/2.0/glTFExporter.ts (hunk 1 above)
Comment (deltakosh) [reply serializers/src/glTF/2.0/glTFExporter.ts]: Well yeah I know, you need to build the .d.ts first and then recompile but no worries the CI will take care of it

File: serializers/src/glTF/2.0/glTFExporter.ts [hunk 3]
```diff
@@ -335,7 +335,11 @@ export class _Exporter {
             localCanvas.id = "WriteCanvas";
             localCanvas.width = 2048;
             localCanvas.height = 2048;
-            this._localEngine = new Engine(localCanvas, true, { premultipliedAlpha: false, preserveDrawingBuffer: true });
+            if (Tools.IsSafari()) {
+                this._localEngine = new Engine(localCanvas, true, { premultipliedAlpha: true, preserveDrawingBuffer: true });
```
Comment (deltakosh) [inline serializers/src/glTF/2.0/glTFExporter.ts]: can we just do something like
```
 this._localEngine = new Engine(localCanvas, true, { premultipliedAlpha: Tools.IsSafari(), preserveDrawingBuffer: true });
```

File: src/Misc/tools.ts (hunk 2 above)
Comment (deltakosh) [reply src/Misc/tools.ts]: We need to keep thinengine with no dependency so let's keep it like that for now

**Comment by deltakosh** [comment]
Also CI is failing big time ;)

File: serializers/src/glTF/2.0/glTFExporter.ts (hunk 1 above)
Comment (Drigax) [reply serializers/src/glTF/2.0/glTFExporter.ts]: Looks like the CI is failing for the same reason:
https://dev.azure.com/babylonjs/ContinousIntegration/_build/results?buildId=3030&view=logs&j=182f86de-37f5-5af4-c789-e92a7c8fa3dc&t=14c2bff8-fe23-5e9c-07aa-9b09ac155f63&l=169

I'm not sure what I'm doing wrong here

Comment (deltakosh) [reply serializers/src/glTF/2.0/glTFExporter.ts]: Are you sure this is the correct Tools (the one from bjs?)

Comment (Drigax) [reply serializers/src/glTF/2.0/glTFExporter.ts]: I made the change to src/Misc/Tools.ts...

Nevermind I'm just being silly. I accidentally put the change as a part of AsyncLoop

File: serializers/src/glTF/2.0/glTFExporter.ts (hunk 3 above)
Comment (Drigax) [reply serializers/src/glTF/2.0/glTFExporter.ts]: sure, made the appropriate change.

**Pull request merged**